extensions/avro/runtime/src/main/java/org/apache/camel/quarkus/component/avro/AvroSchemaRegistry.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.avro;
+
+import java.util.HashMap;
+
+import org.apache.avro.Schema;
+
+public class AvroSchemaRegistry extends HashMap<String, Schema> {
```

Comment:
I have some concerns about this. Let me write more tomorrow.

Reply:
I was about to say that as long as we need just Map.get() and Map.put(), we should prefer aggregation to inheritance. But in the end, can't we fully avoid the AvroSchemaRegistry class in this specific case and use a stock HashMap instead?

Reply:
The AvroSchemaRegistry class can't be avoided due to [#9446](https://github.com/quarkusio/quarkus/issues/9446) however aggregation looks feasible.

Reply:
#9446 can be worked around using a custom `ObjectSubstitution<Schema, byte[]>`. There is no need to pass a HashMap to the recorder, IMO.

Reply:
Something like this: https://github.com/ppalaga/camel-quarkus/commit/5d03784f8907b045188f46ec9aaeebee20b9018a#diff-9b69190fd7df50ae0801559a77503de0R34

Reply:
Ok I see, indeed we can remove the AvroSchemaRegistry level if we pass schemas one by one in the recorder. It's less code and I think startup time will not be significantly impacted. So I'm ok.